test/e2e/metering_manual_install_tests.go

```diff
@@ -52,14 +56,23 @@ func testManualMeteringInstall(
 		meteringOperatorImageTag,
 		reportingOperatorImageRepo,
 		reportingOperatorImageTag,
+		catalogSourceName,
+		catalogSourceNamespace,
+		subscriptionChannel,
 		testCaseOutputBaseDir,
 		testInstallFunction.ExtraEnvVars,
 		mc.Spec,
 	)
 	require.NoError(t, err, "creating a new deployer context should produce no error")
-
 	deployerCtx.Logger.Infof("DeployerCtx: %+v", deployerCtx)
-	rf, err := deployerCtx.Setup(deployerCtx.Deployer.Install, expectInstallErr)
+
+	installFunc := deployerCtx.Deployer.InstallOLM
+	uninstallFunc := deployerCtx.Deployer.UninstallOLM
+	if installationMethod != "olm" {
+		installFunc = deployerCtx.Deployer.Install
+		uninstallFunc = deployerCtx.Deployer.Uninstall
+	}
```

Comment:
This is something we start addressing in the near future, with the intent to provide more flexibility to the e2e suite. At the moment, we pass a closure to the `deployerCtx.Setup` method that expects a `func() error` type signature, but this doesn't account for cases where we expect an error may occur when attempting to create an invalid resource, e.g. a MeteringConfig CR missing a storage specification, or we need to do some pre-processing before installing Metering.

As a short-term workaround, we may be able to just pass a global flag, or a per-test-table-indice installation method to help alleviate this. The cleanest solution is most likely making the registry image a required e2e field, which would also eliminate the need to do this kind of check.

Reply:
And just to tack on to that last comment and provide more context on the `... MeteringConfig CR missing a storage specification`, we shouldn't need to instantiate a deploy-metering context to test out our CRD schema validation as that's a waste of time and pings the api-server more than necessary. We could, instead, update the `deployerCtx.Setup` method to support the different tests that we may want to perform. In that case, we could instead, pass a closure that attempts to create the invalid `MeteringConfig` custom resource, and performs a check to whether or not an error was returned.